### PR TITLE
JRuby compatibility

### PR DESCRIPTION
**Pull request by guizmaii**
Hi @craiglittle,

We're migrating an app using `biz` to JRuby and we saw that JRuby doesn't work with these `BIG_BANG` and `HEAT_DEATH` values.

Do we really need such huge values ?

Here, for `BIG_BANG`, I used the default values used in Ruby when creating a new date via `Date.new`: `-4712` and it's opposite value for `HEAT_DEATH`.

What do you think about this change ?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,11 @@ rvm:
   - 2.3.7
   - 2.4.4
   - 2.5.1
+  - jruby-9.1
+  # - jruby-9.2 # Seems to be bugged. Related issue: https://github.com/jruby/jruby/issues/5320
```
Comment (craiglittle) [inline .travis.yml]: Per the advice in the linked issue (thanks for filing that!), let's change this to `jruby-head`.

File: .travis.yml [hunk 2]
```diff
@@ -4,6 +4,11 @@ rvm:
...
+
+jdk:
+- oraclejdk8
```
Comment (craiglittle) [inline .travis.yml]: Let's remove this and roll with what Travis uses as the default on any given run.

File: lib/clavius/time.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@
 module Clavius
   module Time
     WEEKDAYS   = %i[sun mon tue wed thu fri sat].freeze
-    BIG_BANG   = Date.new(-10**100)
-    HEAT_DEATH = Date.new(10**100)
+    BIG_BANG   = Date.new(-4712, 1, 1, Date::ITALY)
+    HEAT_DEATH = Date.new(4712, 1, 1, Date::ITALY)
```
Comment (craiglittle) [inline lib/clavius/time.rb]: I've done some playing around with these values, and it looks like `100_000_000` is the largest round order of magnitude that works with JRuby, so let's go with `-100_000_000` and `100_000_000`. (Please use the underscores.)

**Review by craiglittle — COMMENTED** [review]
Please address my comments and squash into a single commit, and we'll get this merged!

**Comment by guizmaii** [comment]
> squash into a single commit 

Done but I messed up a bit I think.

File: lib/clavius/time.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@
 module Clavius
   module Time
     WEEKDAYS   = %i[sun mon tue wed thu fri sat].freeze
-    BIG_BANG   = Date.new(-10**100)
-    HEAT_DEATH = Date.new(10**100)
+    BIG_BANG   = Date.new(-100_000_000, 1, 1, Date::ITALY)
+    HEAT_DEATH = Date.new(100_000_000, 1, 1, Date::ITALY)
```
Comment (craiglittle) [inline lib/clavius/time.rb:7]: Just with the year, please :smile::
```ruby
BIG_BANG   = Date.new(-100_000_000)
HEAT_DEATH = Date.new(100_000_000)
```

**Comment by craiglittle** [comment]
Yeah, it looks pretty messed at the moment. I'd suggest resetting your branch to the tip of `master` and applying your changes again since there aren't that many.

**Comment by guizmaii** [comment]
Done here #5

**Pull request closed without merging**